src/shared/protocol.ts

```diff
@@ -332,6 +398,23 @@ export abstract class Protocol<
         }
       });
 
+      options?.signal?.addEventListener("abort", () => {
+        const reason = options?.signal?.reason;
+        this._responseHandlers.delete(messageId);
+        this._progressHandlers.delete(messageId);
+
+        this._transport?.send({
+          jsonrpc: "2.0",
+          method: "cancelled",
```

Comment:
I think this must be `notifications/cancellation`. I took the liberty of fixing this in https://github.com/modelcontextprotocol/typescript-sdk/pull/76.